### PR TITLE
Refactor transactions - Closes #1432

### LINKED ISSUE #1432: Refactor transaction-* database repositories
This will be done in two separate PR-s:

 - [x] **Structure/Basic Refactoring**
 - [x] **Logic Refactoring**

### Structure/Basic Refactoring

* Converting into ES6 classes
* Moving simple/static SQL into external repositories
* Improving use of the `pg-promise`  API
* Documentation for code/sql

### Logic Refactoring

* Moving complex/dynamic SQL into external files where possible
* Improving the main underlying logic - `save` operation
* Simplifying the overall class protocol
* Reorganizing files/folders
* Performance optimizations
* Adding tests

For example, all `repos/transactions-*.js` files can possibly be united. They each implement only one specific method `save` - that's an interface/driver pattern, and not repository. At the very least, they can be moved into a single folder, with a simpler protocol.

### PR DESCRIPTION
**Pull request by vitaly-t**
Closes #1432

Second stage of refactoring transactions.

### CODE REVIEW

**Comment by vitaly-t** [comment]
Parking it for now, as it will create way too many conflicts with #1720

Once that PR has been merged, will rebase this one.

**Comment by vitaly-t** [comment]
Merged with the code in PR #1720, in order to be able to continue working, as otherwise it would create endless conflicts.

**Comment by vitaly-t** [comment]
**IMPORTANT**

This PR is cumulative, and includes PR-s #1718 and #1720. It was done to avoid unresolvable conflicts between this PR and PR #1720, and to be able to finish it while I had time.

File: db/repos/accounts.js [hunk 1]
```diff
@@ -255,7 +255,8 @@ class AccountsRepository {
 	 * @todo Add description for the return value
 	 */
 	insert(data) {
-		return this.db.none(this.pgp.helpers.insert(data, this.cs.insert));
+		const query = () => this.pgp.helpers.insert(data, this.cs.insert);
+		return this.db.none(query);
 	}
```
Comment (nazarhussain) [inline db/repos/accounts.js]: @vitaly-t You made same change into another PR. 

https://github.com/LiskHQ/lisk/pull/1720/files#diff-bb3f5b06bf2cf765f0688199caad2593R258

Also both of the PR are related tissue state that its a change for transactions, and you are changing the accounts here. I believe this PR contains duplicate code from #1720. Please clean the PR so we can review the changes respective to the issue it is addressing.

Comment (vitaly-t) [reply db/repos/accounts.js]: See my notes above, - this is a cumulative update ;) Once the source PR is merged, the other changes will disappear automatically ;)

**Comment by vitaly-t** [comment]
👍

**Comment by MaciejBaj** [comment]
Thank for the work @vitaly-t and the review @nazarhussain and @4miners. 
The PR closes #1432, #1717, #1720.

**Pull request merged**